WordPress/src/main/java/org/wordpress/android/workers/LocalNotificationWorker.kt

```diff
@@ -0,0 +1,101 @@
+package org.wordpress.android.workers
+
+import android.app.PendingIntent
+import android.content.Context
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.toBitmap
+import androidx.work.CoroutineWorker
+import androidx.work.Data
+import androidx.work.ListenableWorker
+import androidx.work.WorkerFactory
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import org.wordpress.android.R
+import org.wordpress.android.workers.LocalNotification.Type
+
+class LocalNotificationWorker(
+    val context: Context,
+    params: WorkerParameters,
+    private val localNotificationHandlerFactory: LocalNotificationHandlerFactory
+) : CoroutineWorker(context, params) {
+    @Suppress("TooGenericExceptionCaught")
+    override suspend fun doWork(): Result {
+        try {
+            val type = Type.fromTag(inputData.getString(TYPE))
+            val handler = type?.let { localNotificationHandlerFactory.buildLocalNotificationHandler(it) }
+
+            val id = inputData.getInt(ID, -1)
+            val title = inputData.getInt(TITLE, -1)
+            val text = inputData.getInt(TEXT, -1)
+            val icon = inputData.getInt(ICON, -1)
+
+            val pendingIntent = PendingIntent.getActivity(
+                    context,
+                    0,
+                    handler?.buildIntent(context),
+                    PendingIntent.FLAG_CANCEL_CURRENT
+            )
+            val builder = NotificationCompat.Builder(
+                    context,
+                    context.getString(R.string.notification_channel_normal_id)
+            )
+                    .setContentIntent(pendingIntent)
+                    .setSmallIcon(icon)
+                    .setContentTitle(context.getString(title))
+                    .setContentText(context.getString(text))
+                    .addAction(R.drawable.ic_story_icon_24dp, context.getString(R.string.cancel),
+                            pendingIntent)
+                    .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+                    .setCategory(NotificationCompat.CATEGORY_REMINDER)
```

Comment:
In the following PR we should reconsider if we need these flags. I've originally introduced only because I was trying what I can set. The same goes to the `addAction` where the icon and the text should probably be part of the `inputData`. We should also remove the `setLargeIcon` which should probably be done in this PR as it doesn't make sense here.

Reply:
I've moved icon and title to inputData and removed setLargeIcon.  Left the priority and category for now, we can remove them if not required when testing the reminders.  Thanks